src/hotspot/share/code/nmethod.hpp

```diff
@@ -463,6 +470,12 @@ class nmethod : public CodeBlob {
   void oops_do_set_strong_done(nmethod* old_head);
 
 public:
+  enum ResultStatus {
+    passed,
+    code_cache_full,
+    out_of_memory
```

Comment:
Maybe `out_of_malloc_memory` would be clearer? Or `out_of_immutable_data_memory` if immutable data may not always be malloc'ed.

Reply:
May be `no_space_for_mmutable_data`.